dev/benchreport/report.go

```diff
@@ -0,0 +1,153 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package benchreport
+
+import (
+	"bytes"
+	"fmt"
+	"math"
+	"os"
+	"text/template"
+)
+
+const (
+	reportExt = ".xml"
+	tpl       = `### :rocket: Benchmarks report
+{{range $package, $reports := .}}
+#### Package ` + "`" + `{{$package}}` + "`" + `
+
+Data stream | Previous EPS | New EPS | Diff (%) | Result
+----------- | ------------ | ------- | -------- | ------
+{{range $reports}}` + "`" + `{{.DataStream}}` + "`" + ` | {{.Old}} | {{.New}} | {{.Diff}} ({{.Percentage}}%) | {{getResult .Old .Percentage}}
+{{end}}{{end}}`
+)
+
+type options struct {
+	sourceDir string
+	targetDir string
+	threshold float64
+}
+
+func (o *options) validate() error {
+	if _, err := os.Stat(o.sourceDir); err != nil {
+		return fmt.Errorf("stat file failed (path: %s): %w", o.sourceDir, err)
+	}
+
+	if _, err := os.Stat(o.targetDir); err != nil {
+		return fmt.Errorf("stat file failed (path: %s): %w", o.targetDir, err)
+	}
+
+	return nil
+}
+
+type report struct {
+	Package    string
+	DataStream string
+	Old        float64
+	New        float64
+	Diff       float64
+	Percentage float64
+}
+
+func GetBenchReport(sourceDir, targetDir string, threshold float64) ([]byte, error) {
```

Comment:
Nit. Mention the kind of report that is being generated here.

Something like:
```suggestion
func GetGithubMarkdownBenchReport(sourceDir, targetDir string, threshold float64) ([]byte, error) {
```